rules.json

```diff
@@ -0,0 +1,20 @@
+{
+    "Slack Token": "(xox[p|b|o|a]-[0-9]{12}-[0-9]{12}-[0-9]{12}-[a-z0-9]{32})",
+    "RSA private key": "-----BEGIN RSA PRIVATE KEY-----[^[]",
+    "SSH (OPENSSH) private key": "-----BEGIN OPENSSH PRIVATE KEY-----[^[]",
+    "SSH (DSA) private key": "-----BEGIN DSA PRIVATE KEY-----[^[]",
+    "SSH (EC) private key": "-----BEGIN EC PRIVATE KEY-----[^[]",
+    "PGP private key block": "-----BEGIN PGP PRIVATE KEY BLOCK-----[^[]",
+    "Facebook Oauth": "[f|F][a|A][c|C][e|E][b|B][o|O][o|O][k|K].*['|\"][0-9a-f]{32}['|\"]",
+    "Twitter Oauth": "[t|T][w|W][i|I][t|T][t|T][e|E][r|R].*['|\"][0-9a-zA-Z]{35,44}['|\"]",
+    "GitHub": "[g|G][i|I][t|T][h|H][u|U][b|B].*['|\"][0-9a-zA-Z]{35,40}['|\"]",
+    "Google Oauth": "(\"client_secret\":\"[a-zA-Z0-9-_]{24}\")",
+    "AWS API Key": "AKIA(?!IOSFODNN7EXAMPLE)[0-9A-Z]{16}",
+    "Heroku API Key": "[h|H][e|E][r|R][o|O][k|K][u|U].*[0-9A-F]{8}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{12}",
+    "Generic Secret": "[s|S][e|E][c|C][r|R][e|E][t|T].*['|\"][0-9a-zA-Z]{32,45}['|\"]",
```

Comment:
This regex makes no sense to me. Either I am the idiot or the person who wrote them.

`[s|S]` matches `s`, `S` or `|`. Is that really the intention?

Reply:
No, you're not the idiot. I think they meant (s|S).

Reply:
Well, my confidence in this solution is pretty much gone. How can I trust a piece of security software if their authors can't even get a regular expression right?